specs/agents/tracing-spans-service-target.md

```diff
@@ -0,0 +1,228 @@
+  if (!service_target.name) { // infer name from span attributes
+      
+    if (span.context.db.instance) {  // database spans
+      service_target.name = span.context.db.instance;
+
+    } else if (span.context.message) { // messaging spans
+      service_target.name = span.context.message.queue.name
+
+    } else if (context.http.url) { // http spans
+      service_target.name = context.http.host;
+      if (context.http.url.port > 0) {
+        service_target.name += ":" + context.http.port;
+      }
+    }
+
+  }
+} else {
+    // non-exit spans should not have service.target.* fields
+    span.context.service.target = undefined;
+}
+```
+
+The values for `span.context.db.instance` are described in [SQL Databases](./tracing-instrumentation-db.md#sql-databases).
+
+The values for `span.context.message.queue.name` are described in [Messaging context fields](./tracing-instrumentation-messaging.md#context-fields)
+
+### User API
+
+Agents SHOULD provide an API entrypoint to set the value of `span.context.destination.service.resource`,
+setting an empty or `null` value allows the user to discard the inferred value.
+This API entrypoint should be marked as deprecated and replaced by the following:
+
+Agents SHOULD provide an API entrypoint to set the value of `span.context.service.target.type` and `span.context.service.target.name`,
+setting an empty or `null` value on both of those fields allows the user to discard the inferred values.
+
+When a user-provided value is set, it should take precedence over inferred values from the span `_.type` `_.subtype` or any `_.context` attribute.
+
+### Phase 3
+
+In Phase 3 the `span.service.target.{type,name}` fields are inferred on APM Server with the following algorithm and internal
+usage of `resource` field in apm-server can be replaced with `span.service.target.{type,name}` fields.
+
+```javascript
+
+// Infer new fields values from an existing 'resource' value
+// Empty type value (but not null) that can be used on UI to use the existing resource for display.
+// For internal aggregation on (type,name) and usage this will be equivalent to relying on 'resource' value.
+inferFromResource = function (r) {
+
+    singleSlashRegex = new RegExp('^([a-z]+)/(\w+)$').exec(r);
+    typeOnlyRegex = new RegExp(('^[a-z]+$')).exec(r);
+
+    if (singleSlashRegex != null) {
+        // Type + breakdown
+        // e.g. 'mysql/mydatabase', 'rabbitmq/myQueue'
+        return {
+            type: singleSlashRegex[1],
+            name: singleSlashRegex[2]
+        }
+
+    } else if (typeOnlyRegex != null) {
+        // Type only
+        // e.g. 'mysql'
+        return {
+            type: r,
+        };
+
+    } else {
+        // Other cases, should rely on default, UI will have to display resource as fallback
+        // e.g. 'localhost:8080'
+
+        return {
+            type: '',
+            name: r
+        }
+    }
+}
+
+// usage with span from agent intake
+span = {};
+
+if (!span.service.target.type) {
```

Comment:
To indicate that inferring the new fields is only relevant when the old field is present - basically only for exit spans.

```suggestion
if (!span.service.target.type && span.destination.service.resource) {
```